File: Backend/src/main/java/de/ali/shitpostbot/Template/model/Template.java
/*
Zur diskussion steht:
Willst du die preview bilder uebernehmen im base64 format?
Willst du nur Koordinaten uebernehmen und draufzeichen?
Draufzeichnen erfolgt via Java.
-> Mit verlinkung zum originalen bild?
 */

package de.ali.shitpostbot.Template.model;

import de.ali.shitpostbot.Coordinate.model.Coordinate;
import de.ali.shitpostbot.User.model.User;
import de.ali.shitpostbot.shared.model.BaseEntity;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import com.fasterxml.jackson.annotation.JsonIgnore;
import javax.persistence.*;
import java.util.Set;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Template extends BaseEntity {
    /**
     * Das originale Bild
     */
    private String baseUrl;

    /**
     * Da wo andere Bilder eingefügt werden
     */
    @JsonIgnore
    @OneToMany(mappedBy="reference", cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<Coordinate> coordinates;

    @ManyToOne
    @JoinColumn(nullable = false)
    @EqualsAndHashCode.Exclude
    private User poster;
}

File: Frontend/src/shared/template/model/Template.ts
import {BaseEntity} from '../../models/BaseEntity';
import {Coordinate} from '../../coordinate/model/Coordinate';
import {User} from '../../user/model/User';

export interface Template extends BaseEntity {
  poster: User;
  baseUrl: string;
}

/*export class Template extends BaseEntity {
  public poster: User;
  public baseUrl: string;

  /*public set baseUrl(baseUrl: string) {
    this._baseUrl = baseUrl;
  }
  public get baseUrl(): string {
    return this._baseUrl;
  }
  public set poster(poster: User){
    this._poster = poster;
  }
  public get poster(): User {
    return this._poster;
  }
}*/
